src/display.rs

```diff
@@ -143,80 +162,67 @@ fn inner_display_tree(
     flags: &Flags,
     colors: &Colors,
     icons: &Icons,
-    depth: usize,
-    prefix: &str,
-) -> String {
-    let mut output = String::new();
+    tree_depth_prefix: (usize, &str),
+    padding_rules: &HashMap<Block, usize>,
+    tree_index: usize,
+) -> Vec<Cell> {
+    let mut cells = Vec::new();
     let last_idx = metas.len();
 
-    let padding_rules = get_padding_rules(&metas, flags);
-
-    let mut grid = Grid::new(GridOptions {
-        filling: Filling::Spaces(1),
-        direction: Direction::LeftToRight,
-    });
+    for (idx, meta) in metas.iter().enumerate() {
+        let current_prefix = if tree_depth_prefix.0 > 0 {
+            if idx + 1 != last_idx {
+                // is last folder elem
+                format!("{}{} ", tree_depth_prefix.1, EDGE)
+            } else {
+                format!("{}{} ", tree_depth_prefix.1, CORNER)
+            }
+        } else {
+            tree_depth_prefix.1.to_string()
+        };
 
-    for meta in metas.iter() {
         for block in get_output(
             &meta,
             &colors,
             &icons,
             &flags,
             &DisplayOption::FileName,
             &padding_rules,
+            (tree_index, &current_prefix),
         ) {
             let block_str = block.to_string();
 
-            grid.add(Cell {
+            cells.push(Cell {
                 width: get_visible_width(&block_str),
                 contents: block_str,
             });
         }
-    }
-
-    let content = grid.fit_into_columns(flags.blocks.0.len()).to_string();
-    let mut lines = content.lines();
-
-    for (idx, meta) in metas.iter().enumerate() {
-        let is_last_folder_elem = idx + 1 != last_idx;
-
-        if depth > 0 {
-            output += prefix;
-
-            if is_last_folder_elem {
-                output += EDGE;
-            } else {
-                output += CORNER;
-            }
-            output += " ";
-        }
-
-        output += &String::from(lines.next().unwrap());
-        output += "\n";
 
         if meta.content.is_some() {
-            let mut new_prefix = String::from(prefix);
-
-            if depth > 0 {
-                if is_last_folder_elem {
-                    new_prefix += LINE;
+            let new_prefix = if tree_depth_prefix.0 > 0 {
+                if idx + 1 != last_idx {
+                    // is last folder elem
+                    format!("{}{}", tree_depth_prefix.1, LINE)
```

Comment:
Just noticed that we could make one tiny change.

```diff
diff --git a/src/display.rs b/src/display.rs
index a8b474c..132341b 100644
--- a/src/display.rs
+++ b/src/display.rs
@@ -202,9 +202,9 @@ fn inner_display_tree(
             let new_prefix = if tree_depth_prefix.0 > 0 {
                 if idx + 1 != last_idx {
                     // is last folder elem
-                    format!("{}{}", tree_depth_prefix.1, LINE)
+                    format!("{}{} ", tree_depth_prefix.1, LINE)
                 } else {
-                    format!("{}{}", tree_depth_prefix.1, BLANK)
+                    format!("{}{} ", tree_depth_prefix.1, BLANK)
                 }
             } else {
                 tree_depth_prefix.1.to_string()
```


|Before|After|
|---|---|
|![Screenshot 2021-03-23 at 10 59 08 PM](https://user-images.githubusercontent.com/14259816/112191002-8f43ce00-8c2b-11eb-94ae-eb29386c5ee0.png)|![Screenshot 2021-03-23 at 10 57 01 PM](https://user-images.githubusercontent.com/14259816/112191038-9965cc80-8c2b-11eb-83ce-350f0ed4afaa.png)|

*Check the tree thing under `meta`*

Reply:
I agree that it would be better, especially with folder icons. :+1: